src/base/static/libs/maps/mapboxgl-provider.js

```diff
@@ -1041,7 +1042,12 @@ export default (container, options) => {
     },
 
     getGeolocateControl: () => {
-      return new mapboxgl.GeolocateControl();
+      geolocateControl = new mapboxgl.GeolocateControl();
+      return geolocateControl;
+    },
```

Comment:
Having a getter that sets a global variable and calls on another constructor seems like an anti-pattern.

Perhaps we can update MainMap to replace:
`this._map.addControl(this._map.getGeolocateControl(), "top-left");` 
with a new function, like `addGeolocationControl`:
`this._map.addGeolocationControl("top-left");`

where `addGeolocationControl` is defined in the provider, like:

```javascript
addGeolocationControl = (position) => {
  geolocateControl = new mapboxgl.GeolocateControl()
  map.addControl(geolocateControl, position)
}
```

Reply:
Ok, that makes sense.